agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -335,6 +380,16 @@ func (state *DockerTaskEngineState) AddContainer(container *apicontainer.DockerC
 		state.tasks[task.Arn] = task
 	}
 
+	_, pulledExist := state.taskToPulledContainer[task.Arn]
+	if pulledExist {
+		seelog.Debug("Delete a pulled container named %s from the pulled container map associated with the task ARN %s since AddContainer is called",
+			container.Container.Name, task.Arn)
+		delete(state.taskToPulledContainer[task.Arn], container.Container.Name)
```

Comment:
we should add unit tests validating the remove.

Reply:
`TestPulledContainerToAddContainer` is added to dockerstate_test.go to validate this behavior . Thanks!